lib/book.py

```diff
@@ -125,117 +112,34 @@ def formats_not_failing(self):
             if info['status'] is not self.FAIL:
                 yield info
 
-    # get book's title, title sort, author, author sort, and asin if it exists
+    # get book's title, title sort, author and author sort if it exists
     def _get_basic_information(self):
         self._title = self._db.field_for('title', self._book_id)
-        self._title_sort = self._db.field_for('sort', self._book_id)
 
         self._author = self._db.field_for('authors', self._book_id)
         if len(self._author) > 0:
             self._author = ' & '.join(self._author)
-        self._author_sort = self._db.field_for('author_sort', self._book_id)
-        if self._title is 'Unknown' or self._title_sort is 'Unknown' or not self._author or not self._author_sort:
+        if self._title is 'Unknown' or not self._author:
             self._status = self.FAIL
             self._status_message = self.FAILED_BASIC_INFORMATION_MISSING
             return
 
-        self._asin = self._book_settings.prefs['asin'] if self._book_settings.prefs['asin'] != '' else None
-        if not self._asin:
-            identifiers = self._db.field_for('identifiers', self._book_id)
-            self._asin = self._db.field_for('identifiers', self._book_id)['mobi-asin'].decode('ascii') if 'mobi-asin' in identifiers.keys() else None
-            if self._asin:
-                self._book_settings.prefs['asin'] = self._asin
-
-        self._shelfari_url = self._book_settings.prefs['shelfari_url'] if self._book_settings.prefs['shelfari_url'] != '' else None
+        self._author_list = list(self._author) if isinstance(self._author, tuple) else self._author
```

Comment:
Yup. This was messy to figure out. The db returns a string if it's a single author and a tuple if it has multiple authors. The reason I changed it to a list was that at first in xray_creator, I was doing "if author != b.author_list" -- this wouldn't work if one was a list and the other a tuple. It turns out that for some reason, getting the author data from the book on the device doesn't always return the full list of authors. I think we should investigate that some more. Maybe I was doing something wrong -- maybe I uploaded the book with only one author then modified it in Calibre after or something.